### PR TITLE
New mogul menu

### PR DESCRIPTION
**Pull request by machupicchu27**
This is the request I'm using to get the state of the controls on the screen. Will probably change to make it more efficient for fewer emits and the element also needs to be changed for fullscreen controls.

### CODE REVIEW

**Comment by machupicchu27** [comment]
Depends on chonk [mk-jumper-attribute-mutation-support](https://github.com/trufflehq/chonk/tree/mk-jumper-attribute-mutation-support] to work

File: stream-projects/mogul-menu/components/menu/extension-icon/extension-icon.tsx [hunk 1]
```diff
@@ -1,36 +1,118 @@
...
     toggleOpen();
   };
 
   useStyleSheet(stylesheet);
   const iconImageObj = getMenuIconImageObj(menuState);
+
+
+
+
+      // console.log(hideMenu)
```
Comment (AMFTech512) [inline stream-projects/mogul-menu/components/menu/extension-icon/extension-icon.tsx:32]: Do we still need all of this commented code?

File: stream-projects/mogul-menu/components/menu/extension-icon/extension-icon.tsx [hunk 2]
```diff
@@ -1,36 +1,66 @@
...
+    getSrcByImageObj, jumper,
+    React,
+    Ripple, useEffect, useState,
+    useStyleSheet,
 } from "../../../deps.ts";
 import { getHasNotification, useTabs } from "../../tabs/mod.ts";
-import { getMenuIconImageObj, useMenu } from "../mod.ts";
+import { getMenuIconImageObj, useMenu, getIsOpen } from "../mod.ts";
 import stylesheet from "./extension-icon.scss.js";
+import hamburgerIcon from "../../../assets/hamburgerIcon.png";
```
Comment (austinhallock) [inline stream-projects/mogul-menu/components/menu/extension-icon/extension-icon.tsx:11]: this won't actually work in prod. vite (what we use in truffle-cli dev) can handle this, but in prod in needs to be pure unbundled js

what we usually do for icons is use the Icon component passing in the icon path https://github.com/trufflehq/truffle-packages/blob/master/stream-projects/patreon/components/premium-content-embed/premium-content-embed.tsx#L120

you should be able to get the icon path by exporting as svg in figma, then viewing source of icon and grabbing the path (assuming the icon has 24x24 viewbox, which rachel typically does - let her know if she didn't)

File: stream-projects/mogul-menu/components/menu/extension-icon/extension-icon.tsx [hunk 3]
```diff
@@ -1,36 +1,66 @@
...
-      }}
-      onClick={onExtensionIconClick}
-    >
-      <Ripple color="var(--mm-color-text-bg-primary)" />
-    </div>
-  );
+    return (
+      <div
+          className={`c-extension-icon ${classKebab({ hasNotification })}`}
+          style={{...(menuOpen && menuOpenStyles)}}
```
Comment (austinhallock) [inline stream-projects/mogul-menu/components/menu/extension-icon/extension-icon.tsx:38]: is this the same as `style={menuOpen && menuOpenStyles}`? or does react freak out if style={false}

Comment (machupicchu27) [reply stream-projects/mogul-menu/components/menu/extension-icon/extension-icon.tsx]: It freaks out if I do that

**Review by AMFTech512 — APPROVED** [review]

**Pull request merged**